com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/v1/SpecialResourceTypesRequiredParameters.java

```diff
@@ -0,0 +1,34 @@
+package com.sap.cloud.lm.sl.cf.core.cf.v1;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import com.sap.cloud.lm.sl.cf.core.model.SupportedParameters;
+
+public enum SpecialResourceTypesRequiredParameters {
+
+    MANAGED_SERVICE_REQUIRED_PARAMETERS(ResourceType.MANAGED_SERVICE, SupportedParameters.SERVICE,
+        SupportedParameters.SERVICE_PLAN), USER_PROVIDED_SERVICE_REQUIRED_PARAMETERS(ResourceType.USER_PROVIDED_SERVICE,
+            SupportedParameters.SERVICE_CONFIG), EXISTING_SERVICE_REQUIRED_PARAMETERS(
+                ResourceType.EXISTING_SERVICE), EXISTING_SERVICE_KEY_REQUIRED_PARAMETERS(ResourceType.EXISTING_SERVICE_KEY);
+
+    private final ResourceType resourceType;
+    private final HashSet<String> requiredParameters = new HashSet<>();
```

Comment:
Use Set<String> as a type for "requiredParameters" (or even Collection<String>) if you don't need any of the HashSet specific methods. This makes it easy to change the used implementation, if necessary (from HashSet to TreeSet, for example).